### PR TITLE
[l10n] Add Danish (da-DK) locale

### PR DESCRIPTION
**Pull request by BossElijah**
Add danish date picker locale

### CODE REVIEW

File: packages/x-date-pickers/src/locales/daDK.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { PickersLocaleText } from './utils/pickersLocaleTextApi';
+import { getPickersLocalization } from './utils/getPickersLocalization';
+
+const daDKPickers: Partial<PickersLocaleText<any>> = {
```
Comment (BossElijah) [inline packages/x-date-pickers/src/locales/daDK.ts:4]: Should I leave this as Partial even though all keys are translated?

**Comment by BossElijah** [comment]
Related issue: #3211

File: packages/x-date-pickers/src/locales/daDK.ts (hunk 1 above)
Comment (LukasTy) [reply packages/x-date-pickers/src/locales/daDK.ts:4]: Based on @alexfauquette opinion—we should leave the `Partial`.
The reason behind it—fewer changes are needed when we are introducing a new label. 😃

File: packages/x-date-pickers/src/locales/daDK.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  fieldDayPlaceholder: () => 'DD',
+  fieldWeekDayPlaceholder: (params) => (params.contentType === 'letter' ? 'EEEE' : 'EE'),
+  fieldHoursPlaceholder: () => 'hh',
+  fieldMinutesPlaceholder: () => 'mm',
+  fieldSecondsPlaceholder: () => 'ss',
+  fieldMeridiemPlaceholder: () => 'aa',
+};
+
+export const DEFAULT_LOCALE = daDKPickers;
+
```
Comment (LukasTy) [inline packages/x-date-pickers/src/locales/daDK.ts:76]: This is only needed and makes sense for `enUS` (default) locale
```suggestion
```

File: packages/x-date-pickers/src/locales/daDK.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  // Toolbar titles
+  datePickerToolbarTitle: 'Vælg dato',
+  dateTimePickerToolbarTitle: 'Vælg dato & tidspunkt',
+  timePickerToolbarTitle: 'Vælg tidspunkt',
+  dateRangePickerToolbarTitle: 'Vælg datointerval',
+
+  // Clock labels
+  clockLabelText: (view, time, adapter) =>
+    `Vælg ${view}. ${
```
Comment (LukasTy) [inline packages/x-date-pickers/src/locales/daDK.ts:35]: Please refer to `deDE`, `frFR`, or any other locale, where the view is translated for an example of how to handle this case.
Currently, the `view` would always be the English word here. 🤔

**Review by LukasTy — COMMENTED** [review]
Thank you for the contribution! 💯 
Leaving a few improvement suggestion. 😉

**Comment by BossElijah** [comment]
Thank you @LukasTy for your comments. I have fixed them now, so in my (humble) opinion, it should be ready for merge – unless you think of something else that could be changed.

**Review by LukasTy — APPROVED** [review]
@BossElijah Thank you once again for your time and effort. LGTM! 👍

**Pull request merged**